tests/by-util/test_whereis.rs

```diff
@@ -0,0 +1,60 @@
+use crate::common::util::TestScenario;
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_basic_lookup() {
+    new_ucmd!()
+        .arg("gcc")
+        .succeeds()
+        .stdout_contains("/usr/bin/gcc");
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_bin_only() {
+    new_ucmd!()
+        .arg("-b")
+        .arg("ping")
+        .succeeds()
+        .stdout_contains("/usr/bin/ping");
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_man_only() {
+    new_ucmd!()
+        .arg("-m")
+        .arg("nmap")
```

Comment:
I don't think `nmap` is a tool that's installed by default and thus this test might fail (it currently fails in the CI).